independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/QueryParamExtractor.java

```diff
@@ -22,27 +18,7 @@ public QueryParamExtractor(String name, boolean single, boolean encoded, String
     }
 
     @Override
-    @SuppressWarnings("unchecked")
     public Object extractParameter(ResteasyReactiveRequestContext context) {
-        Object queryParameter = context.getQueryParameter(name, single, encoded);
-        if (separator != null) {
-            if (queryParameter instanceof List) { // it's List<String>
-                List<String> list = (List<String>) queryParameter;
-                List<String> result = new ArrayList<>(list.size());
-                for (int i = 0; i < list.size(); i++) {
-                    String[] parts = list.get(i).split(separator);
-                    result.addAll(Arrays.asList(parts));
-                }
-                queryParameter = result;
-            } else if (queryParameter instanceof String) {
-                List<String> result = new ArrayList<>(1);
-                String[] parts = ((String) queryParameter).split(separator);
-                result.addAll(Arrays.asList(parts));
-                queryParameter = result;
-            } else {
-                // can't really happen
-            }
-        }
-        return queryParameter;
+        return context.getQueryParameter(name, single, encoded, separator);
```

Comment:
The reason is that we want the separator logic to run when `getQueryParameter` is invoked via the bytecode generated in `ClassInjectorTransformer`.

Another way to achieve this might be to have the separator logic in a separate method somewhere and then have both the `QueryParamExtractor` and the generated bytecode first call `getQueryParameter` and then the method for separator handling.

Reply:
It does look like the right way to do it, IMO.